kytos/core/controller.py

```diff
@@ -355,10 +355,14 @@ def msg_out_event_handler(self):
 
             message = triggered_event.content['message']
             destination = triggered_event.destination
-            if destination:
+            if (destination and
+                    not destination.state == destination.STATES.FINISHED):
                 destination.send(message.pack())
                 self.notify_listeners(triggered_event)
                 self.log.debug("MsgOutEvent handler called")
+            else:
+                self.log.debug(
+                    "connection is finished. Cannot send message")
```

Comment:
It was meant to indicate the 'FINISHED' state, which does not necessarily mean the socket is actually closed.